src/ocl/convolutionocl.cpp

```diff
@@ -479,79 +235,38 @@ void ConvolutionDescriptor::FindConvFwdAlgorithm(Handle& handle,
-    auto ctx = ConvolutionContext{};
-    ctx.SetStream(&handle);
-
-    std::vector<PerfField> perf_db;
-
-    bool use_immediate_solution = false;
-    miopenConvSolution_t sol;
-    if(findMode.IsFast(ctx) || findMode.IsHybrid(ctx))
-    {
-        size_t count;
-        bool fallback;
-        GetForwardSolutions(handle, wDesc, xDesc, yDesc, 1, &count, &sol, &fallback);
-        use_immediate_solution = (count > 0) && !(findMode.IsHybrid(ctx) && fallback);
-        // In Hybrid Find mode, we use Normal Find instead of Immediate fallback kernels.
-    }
-
-    if(use_immediate_solution)
-    {
-        CompileForwardSolution(handle, wDesc, xDesc, yDesc, sol.solution_id);
-        /// It is possible to measure actual execution time and return it to the caller.
-        /// \todo Consider if we need (and want to spend time) for this.
-        const auto id = solver::Id(sol.solution_id);
-        perf_db.push_back(
-            {id.GetAlgo(conv::Direction::Forward), id.ToString(), sol.time, sol.workspace_size});
-    }
-    else
-    {
-        ctx.DetectRocm();
-        ctx.use_dynamic_solutions_only = findMode.IsDynamicHybrid(ctx);
-        perf_db = UserFindDbRecord::TryLoad(handle, problem, [&](DbRecord& record) {
-            DirConvFindCore(handle,
-                            xDesc,
-                            x,
-                            wDesc,
-                            w,
-                            yDesc,
-                            y,
-                            workSpace,
-                            workSpaceSize,
-                            *this,
-                            exhaustiveSearch,
-                            record,
-                            ctx,
-                            problem,
-                            IsWinograd3x3SupportedAndFast(ctx, problem));
-        });
-    }
+    const auto problem =
+        conv::ProblemDescription(xDesc, wDesc, yDesc, *this, conv::Direction::Forward);
+    const auto ctx = [&] {
+        auto tmp = ExecutionContext{&handle};
+        tmp.DetectRocm();
+        problem.SetupFloats(tmp);
+        tmp.do_search = exhaustiveSearch;
+        return tmp;
+    }();
+
+    const auto invoke_ctx = conv::DataInvokeParams{InvokeType::Evaluate,
+                                                   {xDesc, x, wDesc, w, yDesc, y},
+                                                   workSpace,
+                                                   workSpaceSize,
+                                                   attribute.gfx90aFp16alt.GetFwd()};
 
-    if(IsEnabled(MIOPEN_DEBUG_COMPILE_ONLY{}))
-        MIOPEN_THROW(
-            miopenStatusGpuOperationsSkipped,
-            "MIOPEN_DEBUG_COMPILE_ONLY is enabled, escaping forward convolution. Search skipped.");
+    const auto results = FindConvolution(ctx, problem, invoke_ctx);
 
-    if(perf_db.empty())
+    if(results.empty())
         MIOPEN_THROW("Forward Convolution cannot be executed due to incorrect params");
```

Comment:
Can we update this error message to something more descriptive, such as "No suitable algorithm was found to execute the required convolution"

Reply:
Yes, sure.